tests/RobotFramework/tests/cumulocity/telemetry/thin-edge_device_telemetry.robot

```diff
@@ -27,13 +27,13 @@ Thin-edge devices support sending custom measurements
 
 
 Thin-edge devices support sending custom events
-    Execute Command    tedge mqtt pub tedge/events/myCustomType1 '{ "text": "Some test event", "someOtherCustomFragment": {"nested":{"value": "extra info"}} }'
+    Execute Command    tedge mqtt pub te/device/main///e/myCustomType1 '{ "text": "Some test event", "someOtherCustomFragment": {"nested":{"value": "extra info"}} }'
     ${events}=    Device Should Have Event/s    expected_text=Some test event    with_attachment=False    minimum=1    maximum=1    type=myCustomType1    fragment=someOtherCustomFragment
     Log    ${events}
 
 
 Thin-edge devices support sending custom events overriding the type
```

Comment:
What is the false positive / how is it a false positive? Is the `someOtherCustomFragment` property not included in the event?

Reply:
Actually the current main branch has a bug. If user publishes an event with `type` field in the payload, the mapper creates "type" field **twice**. 
* "type":"myCustomType"
* "type":"otherType"

But this test didn't find it because the test checks if contains `type=otherType`.

Here is the trace.
```
[tedge/events/myCustomType] {"type": "otherType", "text": "Some test event", "someOtherCustomFragment": {"nested":{"value": "extra info"}} }
[c8y/event/events/create] {"type":"myCustomType","time":"2023-09-06T09:29:59.89872988Z","text":"Some test event","type":"otherType","someOtherCustomFragment":{"nested":{"value":"extra info"}}}
```

Reply:
> @rina23q So is the fix to avoid duplicated keys also included in the PR?

Yes ofc :) Rust test case `convert_event_use_event_type_from_payload_to_c8y_json()` is covering this.